cumulus/parachains/runtimes/collectives/collectives-westend/src/impls.rs

```diff
@@ -0,0 +1,229 @@
+		length_bound: u32,
+	) -> Result<(u32, u32), DispatchError> {
+		pallet_collective::Pallet::<T, I>::do_propose_proposed(
+			who,
+			threshold,
+			proposal,
+			length_bound,
+		)
+	}
+
+	fn vote_proposal(
+		who: AccountIdOf<T>,
+		proposal: HashOf<T>,
+		index: ProposalIndex,
+		approve: bool,
+	) -> Result<bool, DispatchError> {
+		pallet_collective::Pallet::<T, I>::do_vote(who, proposal, index, approve)
+	}
+
+	fn close_proposal(
+		proposal_hash: HashOf<T>,
+		proposal_index: ProposalIndex,
+		proposal_weight_bound: Weight,
+		length_bound: u32,
+	) -> DispatchResultWithPostInfo {
+		pallet_collective::Pallet::<T, I>::do_close(
+			proposal_hash,
+			proposal_index,
+			proposal_weight_bound,
+			length_bound,
+		)
+	}
+
+	fn proposal_of(proposal_hash: HashOf<T>) -> Option<ProposalOf<T, I>> {
+		pallet_collective::Pallet::<T, I>::proposal_of(proposal_hash)
+	}
+}
+
+/// Used to compare the privilege of an origin inside the scheduler.
+pub struct EqualOrGreatestRootCmp;
+
+impl PrivilegeCmp<OriginCaller> for EqualOrGreatestRootCmp {
+	fn cmp_privilege(left: &OriginCaller, right: &OriginCaller) -> Option<Ordering> {
+		if left == right {
+			return Some(Ordering::Equal)
+		}
+		match (left, right) {
+			// Root is greater than anything.
+			(OriginCaller::system(frame_system::RawOrigin::Root), _) => Some(Ordering::Greater),
+			_ => None,
+		}
+	}
+}
+
+#[cfg(feature = "runtime-benchmarks")]
+pub mod benchmarks {
+	use super::*;
+	use crate::ParachainSystem;
+	use cumulus_primitives_core::{ChannelStatus, GetChannelInfo};
+	use frame_support::traits::{
+		fungible,
+		tokens::{Pay, PaymentStatus},
+	};
+	use pallet_ranked_collective::Rank;
+	use parachains_common::{AccountId, Balance};
+	use sp_runtime::traits::Convert;
+
+	/// Rank to salary conversion helper type.
+	pub struct RankToSalary<Fungible>(PhantomData<Fungible>);
+	impl<Fungible> Convert<Rank, Balance> for RankToSalary<Fungible>
+	where
+		Fungible: fungible::Inspect<AccountId, Balance = Balance>,
+	{
+		fn convert(r: Rank) -> Balance {
+			Balance::from(r).saturating_mul(Fungible::minimum_balance())
+		}
+	}
+
+	/// Trait for setting up any prerequisites for successful execution of benchmarks.
+	pub trait EnsureSuccessful {
+		fn ensure_successful();
+	}
+
+	/// Implementation of the [`EnsureSuccessful`] trait which opens an HRMP channel between
+	/// the Collectives and a parachain with a given ID.
+	pub struct OpenHrmpChannel<I>(PhantomData<I>);
+	impl<I: Get<u32>> EnsureSuccessful for OpenHrmpChannel<I> {
+		fn ensure_successful() {
+			if let ChannelStatus::Closed = ParachainSystem::get_channel_status(I::get().into()) {
+				ParachainSystem::open_outbound_hrmp_channel_for_benchmarks_or_tests(I::get().into())
+			}
+		}
+	}
+
+	/// Type that wraps a type implementing the [`Pay`] trait to decorate its
+	/// [`Pay::ensure_successful`] function with a provided implementation of the
+	/// [`EnsureSuccessful`] trait.
+	pub struct PayWithEnsure<O, E>(PhantomData<(O, E)>);
```

Comment:
I see this already merged in the SDK, no need to copy&past again here.

Reply:
The only other place I see it is in `collectives-polkadot`, which is soon to be gone from the repo. Is this what you mean or is there another type which accomplishes the same thing elsewhere?

Reply:
I've had more of a look and unless you have any objections will leave things as they are, as collectives-polkadot will be removed (hopefully very) soon.